src/main/java/ru/itmo/sd/deadliner2bot/state/AddEndDateState.java

```diff
@@ -21,43 +26,51 @@
 @RequiredArgsConstructor
 public class AddEndDateState implements ChatState {
 
+    private static final ChatStateEnum chatStateEnum = ChatStateEnum.ADD_END_DATE_STATE;
+
     private final ChatRepository chatRepository;
-    private final ChatStateEnum chatStateEnum = ChatStateEnum.ADD_END_DATE;
     private final MessageUtils messageUtils;
     private final DateTimeUtils dateTimeUtils;
     private final TodoService todoService;
+    private final Commands commands;
+    private final StateMessages stateMessages;
+    private Map<String, CommandInfo> commandsInfo;
+
+    @PostConstruct
+    public void postConstruct() {
+        commandsInfo = commands.loadAllCommandsInfo(getChatStateEnum(), List.of(
+                "cancel"
+        ));
+    }
 
     @Override
     public List<BotApiMethod<?>> process(Chat chat, String message) {
         Optional<Todo> todo = todoService.findSelectedTodoByChatId(chat.getChatId());
-        if (message.startsWith("/cancel")) {
+        if (commandsInfo.get("cancel").testMessageForCommand(message)) {
+            chat.setState(ChatStateEnum.EDIT_TODO_STATE);
+            chatRepository.save(chat);
+            return List.of(messageUtils.createMessage(chat, stateMessages.getMessageByKey(chatStateEnum, "cancel")));
+        } else if (message.startsWith("/")) {
+            return null;
+        }
+
+        if (todo.isEmpty()) {
             chat.setState(ChatStateEnum.BASE_STATE);
             chatRepository.save(chat);
-            if (todo.isPresent()) {
-                todo.get().setSelected(false);
-                todoService.save(todo.get());
-            }
```

Comment:
Why did you remove unselection section here? And if you have good reason for that, why didn't remove all unselection sections